.github/send_alerts_to_slack.py

```diff
@@ -0,0 +1,95 @@
+"""
+
+Summarize the Frictionless Data JSON error output and if there are any errors, 
+send a summary message to a given Slack channel. Send a more detailed 
+error summary to std out and create a detailed txt file that can be accessed via GitHub
+artifacts.
+
+To run elsewhere, define SLACK_WEBHOOK in your repo's secrets and change the 
+top-level constants defined below. 
+
+`validate-csv.yml` can also be copied verbatim, assuming that the schema is 
+named `datapackage.json`.
+
+Links to...
+- Generate a Slack token
+    https://slack.com/apps/A0F7XDUAZ-incoming-webhooks
+- API for POST requests / formatting text:
+    https://api.slack.com/messaging
+
+
+"""
+
+import json
+import requests
+import os
+
+ERROR_FILE = 'error_summary.json'
+REPO = 'Princeton-CDH/pemm-data'
+CHANNEL = '#slack-webhook-test'
+ERROR_MAX = 5
+
+def main():
+    with open(ERROR_FILE) as f: 
+        j = json.load(f)
+
+    # Create and print message to std out.
+    stdout_string = f"Error Count: { j['error-count'] }\n"
+    for table in j['tables']:
+        if table['error-count'] != 0:
+            stdout_string += f"\tTable: { table['resource-name'] }\n"
+            stdout_string += f"\tError Count: { table['error-count'] }\n"
+            for error in table['errors'][:ERROR_MAX]:
+                stdout_string += f'\t\t{ error["message"] }\n'
+            stdout_string += "\n"
+
+    print(stdout_string)
+
+    # Create a txt file that users can download as an artifact.
+    with open('error_summary.txt', 'w') as f:
+        f.write(stdout_string)
+
+    # If there were no errors, end function and do not send a Slack message.
+    if j['error-count'] == 0:
+        return
+
+    # Create a slack payload.
+    slack_message_prelude = (
+        f"🚨 Goodtables found {j['error-count']} error(s) in the latest sync." +
+        f" For more details, head to <https://github.com/{REPO}" +
+        f"/actions/runs/{os.environ['GITHUB_RUN_ID']}" + 
+        "?check_suite_focus=true|GitHub Actions>"
```

Comment:
This link doesn't actually open the build log for me. That url looks slightly different (doesn't include `actions`) but the id seems to be different. Any way to link directly to the build?

Reply:
I had confused the run id with the build id. I printed out the environment variables in https://github.com/Princeton-CDH/pemm-data/runs/828223070?check_suite_focus=true if you want to see what's available. (In addition to some googling to see whether I could get the build ID) I could not find the build ID (828223070) in the output of environment variables, but I could find the run ID.

I will fix this, however, to bring the user to the run, but to see what's specifically wrong, they'd have to navigate to the build. It's not ideal, but let me know if it's sufficient.

Reply:
Yes, I see now. If this is the best we can do, then it will be sufficient! It does get you pretty close to the right place. (Seems funny that GitHub didn't think of this use case...)